pkg/support/dump_test.go

```diff
@@ -38,3 +101,97 @@ func TestParseFileName(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, ts.String(), "2021-08-17 09:47:58 +0000 UTC")
 }
+
+func TestTimestampFilter(t *testing.T) {
+	ts := timestampFilter("2m")
+	require.NotNil(t, ts)
+	currentTime := time.Now()
+	expectedTime := currentTime.Add(-2 * time.Minute)
+	assert.WithinDuration(t, expectedTime, *ts, 10*time.Second)
+}
+
+func TestWriteFile(t *testing.T) {
+	resource := "test-data"
+	data := []byte("foo bar")
+
+	fs := afero.NewMemMapFs()
+	err := writeFile(fs, filepath.Join(baseDir, resource), resource, data)
+	require.NoError(t, err)
+
+	fileContents, err := afero.ReadFile(fs, filepath.Join(baseDir, resource))
+	require.NoError(t, err)
+	assert.Equal(t, data, fileContents)
+}
+
+func TestWriteYAMLFile(t *testing.T) {
+	type testData struct {
+		X string
+		Y int
+	}
+	data := testData{
+		X: "foo",
+		Y: 123,
+	}
+	resource := "test-data"
+	path := filepath.Join(baseDir, resource)
+	fs := afero.NewMemMapFs()
+	err := writeYAMLFile(fs, path, resource, &data)
+	require.NoError(t, err)
+
+	fileContents, err := afero.ReadFile(fs, path)
+	require.NoError(t, err)
+	var fileData testData
+	require.NoError(t, yaml.Unmarshal(fileContents, &fileData))
+	assert.Equal(t, data, fileData)
+}
+
+func TestDumpAntctlGet(t *testing.T) {
+	name := "agentinfo"
+
+	exe := new(testExec)
+	expectedOutput := fmt.Sprintf("antctl -oyaml get %s", name)
+
+	fs := afero.NewMemMapFs()
+	err := dumpAntctlGet(fs, exe, name, baseDir)
+	require.NoError(t, err)
+
+	result, _ := afero.ReadFile(fs, "dir1/dir2/agentinfo")
```

Comment:
same comment as above. Use `baseDir` and `filepath.Join` to build the file paths
may apply to other tests as well